### PR TITLE
docs: add explanation on how to use custom APT configuration

### PR DESCRIPTION
**Pull request by renanrodrigo**
## Why is this needed?
This PR solves all of our problems because it documents how to use custom APT configuration in the Pro Client.

### CODE REVIEW

File: docs/howtoguides/use_custom_apt_config.rst [hunk 1]
```diff
@@ -0,0 +1,99 @@
+How to use the Pro Client with custom APT configuration
+*******************************************************
+
+The Pro Client CLI is able, through the security-status command, to show
+information about installed packages in the system, their origins, and
+potential updates that may be available for these. This information is also
```
Comment (s-makin) [inline docs/howtoguides/use_custom_apt_config.rst:6]:
```suggestion
potential updates that may be available for them. This information is also
```

File: docs/howtoguides/use_custom_apt_config.rst [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+available through the
+:ref:`packages updates API<references/api:u.pro.packages.updates.v1>` and the
+:ref:`packages summary API<references/api:u.pro.packages.summary.v1>` calls,
+for example.
+
+The information presented on the outputs mentioned above is taken directly
+from the local APT caches. Packages, origins, versions, candidates, all is
+calculated using the ``python3-apt`` library, which in turn relies on whatever
+APT has knowledge about in a particular machine.
```
Comment (s-makin) [inline docs/howtoguides/use_custom_apt_config.rst:15]:
```suggestion
from the local APT caches. Packages, origins, versions, and candidates are all
calculated using the ``python3-apt`` library, which in turn relies on the knowledge
APT has about in a particular machine.
```

File: docs/howtoguides/use_custom_apt_config.rst [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+Some users may have the need to configure APT to some specific behaviour. An
```
Comment (s-makin) [inline docs/howtoguides/use_custom_apt_config.rst:17]:
```suggestion
You may need to configure APT to have some specific behaviour. An
```

File: docs/howtoguides/use_custom_apt_config.rst [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+example of a use case would be to pass custom ``.list`` or ``.sources`` files,
```
Comment (s-makin) [inline docs/howtoguides/use_custom_apt_config.rst:18]:
```suggestion
example could be to pass custom ``.list`` or ``.sources`` files
```

File: docs/howtoguides/use_custom_apt_config.rst [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+sources.
+
+The Pro Client will respect the `configuration set directly on APT`_, at
+execution time. If system level configuration is present, it will be used by
+default. To configure APT without changing the system, there is the option of
+passing the config file as an environment variable.
+
+Use an APT config file
+=========================
+For example, consider that the ``/etc/apt/apt.conf.d/50-custom`` file exists,
```
Comment (s-makin) [inline docs/howtoguides/use_custom_apt_config.rst:29]:
```suggestion
=========================

For example, consider that the ``/etc/apt/apt.conf.d/50-custom`` file exists,
```

Comment (s-makin) [reply docs/howtoguides/use_custom_apt_config.rst:29]: rST likes to have a blank line after headers :)

File: docs/howtoguides/use_custom_apt_config.rst [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+and has this content:
+
+.. code-block:: bash
+
+    $ cat /etc/apt/apt.conf.d/50-custom
+    # setting custom sources.list and unsetting the parts directory
+    Dir::Etc::Sourcelist "/my/personal/configured/sources.list";
+    Dir::Etc::Sourceparts "nonexisting-dir";
+
+APT will be using this configuration by default when running all operations.
```
Comment (s-makin) [inline docs/howtoguides/use_custom_apt_config.rst:39]:
```suggestion
APT will use this configuration by default when running all operations.
```

File: docs/howtoguides/use_custom_apt_config.rst [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+The Pro Client will do the same - all outputs will take the custom ``.list``
+file into consideration, and no parts directory will be loaded (unless the
```
Comment (s-makin) [inline docs/howtoguides/use_custom_apt_config.rst:41]:
```suggestion
file into consideration, and no ``parts`` directory will be loaded (unless the
```

File: docs/howtoguides/use_custom_apt_config.rst [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+directory set there actually exists ;)
```
Comment (s-makin) [inline docs/howtoguides/use_custom_apt_config.rst:42]:
```suggestion
directory set there actually exists).
```

File: docs/howtoguides/use_custom_apt_config.rst [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+
+Use an environment variable
+===========================
+Now, if the user does not want to change their system, the ``APT_CONFIG``
```
Comment (s-makin) [inline docs/howtoguides/use_custom_apt_config.rst:46]:
```suggestion
===========================

Now, if you do not want to change the system, the ``APT_CONFIG``
```

File: docs/howtoguides/use_custom_apt_config.rst [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+APT, it will also be respected:
+
+.. code-block:: bash
+
+    $ APT_CONFIG=/tmp/custom-config pro security-status
+    $ APT_CONFIG=/tmp/custom-config pro api u.pro.packages.updates.v1
+
+In the python API integration
+=============================
+Changing the environment also works for the integration with the Pro Client's
```
Comment (s-makin) [inline docs/howtoguides/use_custom_apt_config.rst:78]:
```suggestion
=============================

Changing the environment also works for the integration with the Pro Client's
```

File: docs/howtoguides/use_custom_apt_config.rst [hunk 11]
```diff
@@ -0,0 +1,99 @@
+How to use the Pro Client with custom APT configuration
```
Comment (s-makin) [inline docs/howtoguides/use_custom_apt_config.rst:1]: Can we shorten this a bit to something more like "How to use custom APT configuration"?

**Review by s-makin — COMMENTED** [review]
Just some nits :)

File: docs/howtoguides/use_custom_apt_config.rst (hunk 5 above)
Comment (renanrodrigo) [reply docs/howtoguides/use_custom_apt_config.rst:29]: wah rST can mind its own business I write no useless lines

Comment (renanrodrigo) [reply docs/howtoguides/use_custom_apt_config.rst:29]: joking, will fix it.

File: docs/howtoguides/use_custom_apt_config.rst (hunk 8 above)
Comment (renanrodrigo) [reply docs/howtoguides/use_custom_apt_config.rst:42]: typo, but my version ended up better anyway

**Comment by renanrodrigo** [comment]
@s-makin all addressed, thanks for looking!

Link check fails because the security CVE endpoints are so nice

File: docs/howtoguides/use_custom_apt_config.rst [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+    # set the custom configuration, so updates will respect it
+    os.environ["APT_CONFIG"] = "/tmp/custom-config"
+    updates()
+
+    # unsetting the envvar will cause updates to respect system config again
+    os.environ["APT_CONFIG"] = ""
+    updates()
+
+
+.. _configuration set directly on APT: https://manpages.debian.org/unstable/apt/apt.conf.5.en.html
```
Comment (orndorffgrant) [inline docs/howtoguides/use_custom_apt_config.rst:101]: Lets stay within our brand: https://manpages.ubuntu.com/manpages/apt

**Review by orndorffgrant — APPROVED** [review]
LGTM thank you!

File: docs/howtoguides/use_custom_apt_config.rst (hunk 12 above)
Comment (renanrodrigo) [reply docs/howtoguides/use_custom_apt_config.rst]: done

**Comment by renanrodrigo** [comment]
@orndorffgrant I made the edit to the python code - added a `try...finally` block there to the example, let me know if you think it looks better

**Review by dheyay — APPROVED** [review]
:+1:

File: docs/howtoguides/use_custom_apt_config.rst [hunk 13]
```diff
@@ -0,0 +1,103 @@
+How to use custom APT configuration
+***********************************
+
+The Pro Client CLI is able, through the security-status command, to show
```
Comment (s-makin) [inline docs/howtoguides/use_custom_apt_config.rst:4]:
```suggestion
The Pro Client CLI is able, through the ``security-status`` command, to show
```

**Review by lucasmoura — APPROVED** [review]

**Pull request merged**